projects/step-frontend/src/lib/modules/timeseries/dashboard/time-series-dashboard.component.ts

```diff
@@ -0,0 +1,135 @@
+import { AfterViewInit, Component, Input, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { TsFilterItem } from '../performance-view/filter-bar/model/ts-filter-item';
+import { TimeRangePickerSelection } from '../time-selection/time-range-picker-selection';
+import { TimeSeriesContext } from '../time-series-context';
+import { RelativeTimeSelection } from '../time-selection/model/relative-time-selection';
+import { TimeSeriesContextsFactory } from '../time-series-contexts-factory.service';
+import { TimeSeriesDashboardSettings } from './model/ts-dashboard-settings';
+import { PerformanceViewSettings } from '../performance-view/model/performance-view-settings';
+import { TSTimeRange } from '../chart/model/ts-time-range';
+import { PerformanceViewComponent } from '../performance-view/performance-view.component';
+import { RangeSelectionType } from '../time-selection/model/range-selection-type';
+import { filter, Observable, of, Subject, Subscription, takeUntil, combineLatest, throttle, merge, skip } from 'rxjs';
+import { TimeSeriesContextParams } from '../time-series-context-params';
+
+@Component({
+  selector: 'step-timeseries-dashboard',
+  templateUrl: './time-series-dashboard.component.html',
+  styleUrls: ['./time-series-dashboard.component.scss'],
+})
+export class TimeSeriesDashboardComponent implements OnInit, OnDestroy, AfterViewInit {
+  readonly ONE_HOUR_MS = 3600 * 1000;
+
+  @Input() settings!: TimeSeriesDashboardSettings;
+  @ViewChild(PerformanceViewComponent) performanceView!: PerformanceViewComponent;
+
+  timeRangeSelection: TimeRangePickerSelection = { type: RangeSelectionType.FULL };
+  context!: TimeSeriesContext;
+  performanceViewSettings: PerformanceViewSettings | undefined;
+  updateSubscription = new Subscription();
+  update$: Observable<any> | undefined;
+  updateGate$: Subject<any> = new Subject<any>();
+  queuedSubscription = new Subscription();
+  throttledRefreshTrigger$: Subject<any> = new Subject();
+  terminator$ = new Subject();
+
+  timeRangeOptions: RelativeTimeSelection[] = [
+    { label: 'Last 1 Minute', timeInMs: this.ONE_HOUR_MS / 60 },
+    { label: 'Last 5 Minutes', timeInMs: this.ONE_HOUR_MS / 12 },
+    { label: 'Last 30 Minutes', timeInMs: this.ONE_HOUR_MS / 2 },
+    { label: 'Last Hour', timeInMs: this.ONE_HOUR_MS },
+  ];
+
+  constructor(private contextsFactory: TimeSeriesContextsFactory) {}
+
+  ngAfterViewInit(): void {}
+
+  ngOnInit(): void {
+    if (!this.settings) {
+      throw new Error('Settings input must be set');
```

Comment:
is this a common practice in the project? I mean the exclamation mark is used to assert that settings will always exist, it is totally natural for the component to not work if it doesn't, I don't see a point of throwing an error ourselves explicitly

Reply:
I don't this it's a bad idea per se, all I am saying is that I have seen other components and none of them used such an approach, and it is really important for the code to follow the same rules and conventions, otherwise it gets really messed up

Reply:
this is something that should be first discussed with the team and agreed upon as a common approach to follow, and then applied everywhere (consistency), and also be included in the code style guide